src/pages/teachers/curriculum/sitemap.xml/index.ts

```diff
@@ -0,0 +1,65 @@
+import path from "node:path/posix";
+
+import { GetServerSideProps } from "next";
+import { getServerSideSitemap } from "next-sitemap";
+
+import curriculumApi2023 from "@/node-lib/curriculum-api-2023";
+
+/**
+ * Get all curriculum pages and construct sitemap entries for them.
+ */
+
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  const sitemapBaseUrl = process.env.SITEMAP_BASE_URL || "";
+
+  const basePath = "teachers/curriculum";
+
+  const curriculumResult = await curriculumApi2023.subjectPhaseOptions();
+
+  const curriculumPathSlugs: string[] = [];
+
+  for (const curriculum of curriculumResult) {
+    curriculumPathSlugs.push(`${curriculum.slug}-primary`);
```

Comment:
All the curricula cover primary so this is fine for now, but it may not be in the future. I recommend adding a check to make sure the primary phase exists.

Reply:
Now you mention it, there's no primary Music so this logic is already wrong.